RELEASE.md

```diff
@@ -9,45 +9,51 @@ For you to follow along according to these instructions, you need:
   project](https://pypi.org/project/chartpress/).
 - To have push rights to the [chartpress GitHub repository](https://github.com/jupyterhub/chartpress).
 
-## Technical steps to make a release
+## Steps to make a release
 
-1. Checkout master and make sure it is up to date.
+1. Update [CHANGELOG.md](CHANGELOG.md) if it is not up to date,
+   and verify [README.md](README.md) has an updated output of running `--help`.
+   Make a PR to review the CHANGELOG notes.
 
-   ```
+1. Once the changelog is up to date, checkout master and make sure it is up to date and clean.
+
+   ```bash
+   ORIGIN=${ORIGIN:-origin} # set to the canonical remote, e.g. 'upstream' if 'origin' is not the official repo
    git checkout master
-   git fetch <upstream> master
-   git reset --hard <upstream>/master
+   git fetch $ORIGIN master
+   git reset --hard $ORIGIN/master
+   # WARNING! This next command deletes any untracked files in the repo
+   git clean -xfd
    ```
 
-1. Update [CHANGELOG.md](CHANGELOG.md) and verify [README.md](README.md) has an
-   updated output of running `--help`.
-
-1. Set the `__version__` variable in [chartpress.py](chartpress.py)
-   appropriately and make a commit with message `release <tag>`.
+1. Update the version with `bump2version` (can be installed with `pip install -r requirements-dev.txt`)
 
-1. Create a git tag for the commit.
-
-   ```
-   git tag -a <tag> -m <tag> HEAD
+   ```bash
+   VERSION=...  # e.g. 1.2.3
+   bump2version --tag --new-version $VERSION patch
```

Comment:
One of the things I like about bumpversion is it removes the need to calculate version numbers, instead you pass `"major"`, `"minor"`, "`patch"`, "`release`", and it _does the right thing_.

It took me a while to figure out but I think the current config is correct:
1. `bumpversion major`: this indicates the next release will be a major release, version is `MAJOR.0.0.dev`
2. `bumpversion release`: Releases the current version (enable tagging), version is `MAJOR.0.0`
3. `bumpversion patch`: this indicates the next release will be a patch release, version is `MAJOR.0.1.patch`
4. `bumpversion minor`: the next release will be a minor release instead of a patch, version is `MAJOR.1.0.patch`
5. `bumpversion release`: Releases the current version (enable tagging), version is `MAJOR.1.0`

Reply:
I reacted on this as well, got a bit upset about the situation, and wrote https://github.com/jupyterhub/chartpress/pull/74#pullrequestreview-314552099 about this situation.

I currently believe specifying --new-version here is the best approach assuming we want to adhere to the practice of bumping the patch versions and suffixing `.dev` after making a release. I believe not specifying it requires too much insight into bump2version and our configuration of it, to the extent that it isn't even worth using.

Reply:
I read your review!

> I find the practice of bumping the patch version and appending a pre-release of .dev following a release makes it harder to use bump2version in a way that that complies with that practice.

Could you expand on what you dislike about the above workflow? The idea is you shouldn't care about how bumpversion works. You do need to know about how it's using major/minor/patch/release, but if you specify `$VERSION` manually you still need to decide whether it's  a patch/minor/major bump.

Reply:
Understanding the difference between making a path/minor/major bump release is what feels not obvious enough to me. It can be explained though, as caused by the pre-bumped patch version from the latest release made.

When i think of it as an optional additional step if we want to bump major/minor, it feels better in my mind, less complicated. I like this approach best atm.

```
# if you want to first bump major/minor
...

# Now tag the current .dev version!
...
```

Reply:
@manics and @minrk, what about doing what we do in [step 3 of CHP's RELEASE.md](https://github.com/jupyterhub/configurable-http-proxy/blob/master/RELEASE.md) instead of setting a specific version?

This is at least where I've landed that I think is nice.